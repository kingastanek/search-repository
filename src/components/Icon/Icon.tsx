import icons from 'icons/icons';
import COLORS from 'styles/colors';

export type tIcon = {
  icon: string;
  fill?: string;
  width?: string;
  height?: string;
};

const Icon = ({ icon, fill = COLORS.GREY }: tIcon): JSX.Element | null => {
  const IconComponent = icons[icon];

  if (!IconComponent) return null;
  return <IconComponent fill={fill} />;
};

export default Icon;
