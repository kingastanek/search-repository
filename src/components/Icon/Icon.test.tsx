import { shallow } from 'enzyme';
import Icon from './Icon';

describe('Icon', () => {
  const NoIcon = shallow(<Icon icon='_non_existing_icon_name' />);
  it('should return null', () => {
    expect(NoIcon.isEmptyRender()).toBe(true);
  });

  const IconSM = shallow(<Icon icon='search' width='1rem' height='1rem' />);
  it('should render small size correctly', () => {
    expect(IconSM).toMatchSnapshot();
  });

  const IconMD = shallow(<Icon icon='search' />);
  it('should render medium size correctly', () => {
    expect(IconMD).toMatchSnapshot();
  });

  const IconBIG = shallow(<Icon icon='search' width='2rem' height='2rem' />);
  it('should render big size correctly', () => {
    expect(IconBIG).toMatchSnapshot();
  });

  const IconLG = shallow(<Icon icon='search' width='2.5rem' height='2.5rem' />);
  it('should render large size correctly', () => {
    expect(IconLG).toMatchSnapshot();
  });

  const IconRed = shallow(<Icon icon='search' fill='#f00' />);
  it('should render red color fill correctly', () => {
    expect(IconRed).toMatchSnapshot();
  });
});
