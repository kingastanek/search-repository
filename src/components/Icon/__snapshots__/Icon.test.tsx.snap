// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render big size correctly 1`] = `ShallowWrapper {}`;

exports[`Icon should render large size correctly 1`] = `ShallowWrapper {}`;

exports[`Icon should render medium size correctly 1`] = `ShallowWrapper {}`;

exports[`Icon should render red color fill correctly 1`] = `ShallowWrapper {}`;

exports[`Icon should render small size correctly 1`] = `ShallowWrapper {}`;
